flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/filesystem/FileSystemOutputFormat.java

```diff
@@ -85,15 +85,26 @@ private FileSystemOutputFormat(
 
     @Override
     public void finalizeGlobal(int parallelism) {
+        Exception exception = null;
         try {
             FileSystemCommitter committer =
                     new FileSystemCommitter(
                             fsFactory, msFactory, overwrite, tmpPath, partitionColumns.length);
             committer.commitUpToCheckpoint(CHECKPOINT_ID);
         } catch (Exception e) {
-            throw new TableException("Exception in finalizeGlobal", e);
-        } finally {
-            new File(tmpPath.getPath()).delete();
+            exception = new TableException("Exception in finalizeGlobal", e);
+        }
+
+        // delete the temporary directory
```

Comment:
Why not keep this in the finally block?

Reply:
I think when an exception is raised in both `try` and `finally` blocks, the exception in the try block will be lost.

Reply:
I guess we can just catch and discard the exceptions during removing the temp folder (with a warning log perhaps). There's no need to fail the job for it.

Reply:
I get it, thanks.